File: VB/DocumentViewerReportServerClient/Controllers/HomeController.vb
Imports DocumentViewerReportServerClient.Models
Imports System.Web.Mvc

Namespace DocumentViewerReportServerClient.Controllers
    Public Class HomeController
        Inherits Controller

        Public Function Index() As ActionResult
            ViewBag.Title = "Remotely Generated Documents"

            Dim documents = ReportServerHelper.GetReports()
            Return View(documents)
        End Function

        Public Function Viewer(ByVal item As DocumentItem) As ActionResult
            ViewBag.Title = "Preview: " & item.name

            Dim model = New ReportViewerModel() With { _
                .ReportUrl = "report/" & item.id.ToString(), _
                .AuthToken = ReportServerHelper.GetToken(), _
                .ServerUri = ReportServerHelper.REPORT_SERVER_URI _
            }
            Return View(model)
        End Function
    End Class
End Namespace

File: CS/DocumentViewerReportServerClient/Controllers/HomeController.cs
using DocumentViewerReportServerClient.Models;
using System.Web.Mvc;

namespace DocumentViewerReportServerClient.Controllers {
    public class HomeController : Controller {
        public ActionResult Index() {
            ViewBag.Title = "Remotely Generated Documents";

            var documents = ReportServerHelper.GetReports();
            return View(documents);
        }

        public ActionResult Viewer(DocumentItem item) {
            ViewBag.Title = "Preview: " + item.name;

            var model = new ReportViewerModel() {
                ReportUrl = "report/" + item.id.ToString(),
                AuthToken = ReportServerHelper.GetToken(),
                ServerUri = ReportServerHelper.REPORT_SERVER_URI
            };            
            return View(model);
        }
    }
}